### ISSUE TITLE
Build failure with libcurl 8.11

### ISSUE THREAD

**Issue by leejm516** [jeroen/curl#354]
## System info
```
$ uname -a
Linux mbrcoldserver 6.6.59-1-lts #1 SMP PREEMPT_DYNAMIC Fri, 01 Nov 2024 06:29:43 +0000 x86_64 GNU/Linux

```
```
$ sudo pacman -Q | grep -i curl
curl 8.11.0-3
libcurl-compat 8.11.0-3
libcurl-gnutls 8.11.0-3

```

## Error log
```
> BiocManager::install('curl', type = "source", checkBuilt = TRUE)
Bioconductor version 3.20 (BiocManager 1.30.25), R 4.4.2 (2024-10-31)
Installing package(s) 'curl'
trying URL 'https://cloud.r-project.org/src/contrib/curl_6.0.0.tar.gz'
Content type 'application/x-gzip' length 932630 bytes (910 KB)
==================================================
downloaded 910 KB

* installing *source* package ‘curl’ ...
** package ‘curl’ successfully unpacked and MD5 sums checked
** using staged installation
Found pkg-config cflags and libs!
Using PKG_CFLAGS=
Using PKG_LIBS=-lcurl
Found recent recent version of libcurl
** libs
using C compiler: ‘clang version 18.1.8’
rm -f curl.so callbacks.o curl.o download.o escape.o fetch.o findport.o form.o getdate.o handle.o ieproxy.o init.o interrupt.o multi.o nslookup.o options.o reflist.o split.o ssl.o typechecking.o urlparser.o utils.o version.o winidn.o writer.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c callbacks.c -o callbacks.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c curl.c -o curl.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c download.c -o download.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c escape.c -o escape.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c fetch.c -o fetch.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c findport.c -o findport.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c form.c -o form.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c getdate.c -o getdate.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c handle.c -o handle.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c ieproxy.c -o ieproxy.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c init.c -o init.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c interrupt.c -o interrupt.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c multi.c -o multi.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c nslookup.c -o nslookup.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c options.c -o options.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c reflist.c -o reflist.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c split.c -o split.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c ssl.c -o ssl.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c typechecking.c -o typechecking.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c urlparser.c -o urlparser.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c utils.c -o utils.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c version.c -o version.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c winidn.c -o winidn.o
clang -I"/opt/R-4.4.2/lib64/R/include" -DNDEBUG -DSTRICT_R_HEADERS -DR_NO_REMAP  -I/usr/local/include   -fvisibility=hidden -fpic  -march=native -O2 -flto=thin -c writer.c -o writer.o
clang -shared -march=native -O2 -flto=thin -fpic -Wl,-mllvm,-threads=4 -L/opt/R-4.4.2/lib64/R/lib -fuse-ld=lld -Wl,-O2 -Wl,--as-needed -o curl.so callbacks.o curl.o download.o escape.o fetch.o findport.o form.o getdate.o handle.o ieproxy.o init.o interrupt.o multi.o nslookup.o options.o reflist.o split.o ssl.o typechecking.o urlparser.o utils.o version.o winidn.o writer.o -lcurl -L/opt/R-4.4.2/lib64/R/lib -lR
installing to /home/user1/R/x86_64-pc-linux-gnu-library/4.4/00LOCK-curl/00new/curl/libs
** R
** inst
** byte-compile and prepare package for lazy loading
Error in source("tools/option_table.txt", env) :
  tools/option_table.txt:23:20: unexpected input
22:   CURLOPT_HTTPHEADER = 10000 + 23
23:   CURLOPT_HTTPPOST _
                       ^
Error: unable to load R code in package ‘curl’
Execution halted
ERROR: lazy loading failed for package ‘curl’
* removing ‘/home/user1/R/x86_64-pc-linux-gnu-library/4.4/curl’

The downloaded source packages are in
        ‘/tmp/RtmpfhP6SQ/downloaded_packages’
Warning message:
In install.packages(...) :
  installation of package ‘curl’ had non-zero exit status

```

**Comment by jeroen** [comment]
Ugh thanks for the report. There were several breaking changes and regressions in curl 8.11. I'll try to get this fixed. I guess there is no way you can use an older libcurl for now?

**Comment by jeroen** [comment]
This is now fixed in curl 6.0.1 on CRAN.
